xpti/src/xpti_proxy.cpp

```diff
@@ -139,6 +115,38 @@ class ProxyLoader {
     static ProxyLoader *loader = new ProxyLoader();
     return *loader;
   }
+  // needed for testing
+  void tryToEnable() {
```

Comment:
need to discuss with Vasanth. Here it is one way method which could check and enable in runtime. Do we need an opportunity to disable it in runtime too?

Reply:
> need to discuss with Vasanth. Here it is one way method which could check and enable in runtime. Do we need an opportunity to disable it in runtime too?

We seem to be able to enable/disable at runtime already. Currently, setting XPTI_TRACE_ENABLE=0 will disable the instrumentation as xptiTraceEnabled() will fail. Are you referring to something different? Are you referring to being able to do this programmatically?

Reply:
@tovinkere yes, I need to have an option to enable traces in runtime. Now xpti checks env vars at loading and no way to update status after.